File: EXPERIENCIA-DE-PRÁCTICA-2-EN-CAPSULAMIENTO/EJERCICIO-2.2/clase_CuentaBancaria.java
// Clase CuentaBancaria: Representa una cuenta bancaria con atributos como saldo y número de cuenta, y métodos get y set para cada atributo. 

public class CuentaBancaria { 

    // Atributos 

    private double saldo; 

    private String numeroCuenta; 

 

    // Constructor 

    public CuentaBancaria(double saldo, String numeroCuenta) { 

        this.saldo = saldo; 

        this.numeroCuenta = numeroCuenta; 

    } 

 

    // Métodos get para cada atributo 

    public double getSaldo() { // Obtener saldo 

        return saldo; 

    } 

 

    public String getNumeroCuenta() { // Obtener número de cuenta 

        return numeroCuenta; 

    } 

 

    // Métodos set para cada atributo 

    public void setSaldo(double saldo) { // Establecer saldo 

        this.saldo = saldo; 

    } 

 

    public void setNumeroCuenta(String numeroCuenta) { // Establecer número de cuenta 

        this.numeroCuenta = numeroCuenta; 

    } 

 

    // Método main para probar la clase 

    public static void main(String[] args) { 

        // Objetos 

        CuentaBancaria cuenta1 = new CuentaBancaria(1000, "1234567890"); 

        CuentaBancaria cuenta2 = new CuentaBancaria(5000, "0987654321"); 

 

        // Actualizar información utilizando métodos set 

        cuenta1.setSaldo(1500); 

        cuenta2.setNumeroCuenta("9876543210"); 

 

        // Recuperar información utilizando métodos get 

        System.out.println(cuenta1.getSaldo() + " " + cuenta1.getNumeroCuenta()); 

        System.out.println(cuenta2.getSaldo() + " " + cuenta2.getNumeroCuenta()); 

    } 

} 

 

File: EXPERIENCIA-DE-PRÁCTICA-2-EN-CAPSULAMIENTO/EJERCICIO-2.2/ClaseCuentaBancaria.cpp
#include <iostream>
#include <string>

using namespace std;

// clase CuentaBancaria
class CuentaBancaria {
private:
    double saldo;
    string numeroCuenta;

public:
    // establecer los valores de los atributos
    void setSaldo(double _saldo) // establecer el saldo de la cuenta bancaria
    {
        saldo = _saldo;
    }

    void setNumeroCuenta(string _numeroCuenta) // establecer el número de cuenta bancaria
    {
        numeroCuenta = _numeroCuenta;
    }

    // obtener los valores de los atributos
    double getSaldo() const // obtener el saldo de la cuenta bancaria
    {
        return saldo;
    }

    string getNumeroCuenta() const // obtener el número de cuenta bancaria
    {
        return numeroCuenta;
    }
};

int main() 
{
    CuentaBancaria cuenta1, cuenta2;

    // atributos para la cuenta bancaria 1
    cuenta1.setSaldo(9874.0);
    cuenta1.setNumeroCuenta("123456789");

    // atributos para la cuenta bancaria 2
    cuenta2.setSaldo(5211.0);
    cuenta2.setNumeroCuenta("987654321");

    // información sobre las cuentas bancarias
    cout <<"\tCuenta Bancaria 1:" <<endl; // información sobre la cuenta bancaria 1
    cout << "Saldo: S/" << cuenta1.getSaldo() << "\nNumero de Cuenta---> " << cuenta1.getNumeroCuenta() << endl;

    cout <<"\tCuenta Bancaria 2:" <<endl; // información sobre la cuenta bancaria 2
    cout << "Saldo: S/" << cuenta2.getSaldo() << "\nNumero de Cuenta---> " << cuenta2.getNumeroCuenta() << endl;

    return 0;
}
